[MNGSITE-548] Update subproject guide to new term

Bukama opened the pull request:
This PR updates the guide for multi-project setups to the new term "subproject".
It also consistently uses lowercase for the term subproject(s) in headings (except the page header).

elharo reviewed (changes requested):
How do we redirect from the old URL to the new URL?

https://www.w3.org/Provider/Style/URI

cstamas commented:
> How do we redirect from the old URL to the new URL?
> 
> https://www.w3.org/Provider/Style/URI

See https://github.com/apache/maven-site/blob/master/content/resources/xsd/.htaccess
We already do that for XSDs

slawekjaranowski commented:
> > How do we redirect from the old URL to the new URL?
> > https://www.w3.org/Provider/Style/URI
> 
> Readded the old page with a link to the new one.

You should add redirect to: `content/filtered-resources/.htaccess` instead of page

slawekjaranowski commented on content/resources/guides/mini/multi-subprojects.png:
Can we try to use `puml` for it?

Bukama replied:
As my experience with puml is quite limited as of now, I would like to tackle this task while supporting you on #482.

But you are the boss :) I can try to do this after xmas.

slawekjaranowski replied:
picture is not an issues, can be improved later.

Bukama commented:
> > > How do we redirect from the old URL to the new URL?
> > > https://www.w3.org/Provider/Style/URI
> > 
> > 
> > Readded the old page with a link to the new one.
> 
> You should add redirect to: `content/filtered-resources/.htaccess` instead of page

Thanks for the hint, updated (and removed the old one again :D )

elharo commented on content/markdown/guides/mini/guide-multiple-subprojects-4.md:
```diff
@@ -0,0 +1,184 @@
+
+Subprojects collection starts from one aggregator project.
+That project defines the subprojects of which it consists using the `<subproject>` element.
+This is a recursive process, so aggregators can have child subproject which are aggregators themselves.
```
child subprojects

elharo commented on content/markdown/guides/mini/guide-multiple-subprojects-4.md:
```diff
@@ -0,0 +1,184 @@
+
+Because subprojects within a multi-project setup build can depend on each other, it is important that the reactor sorts
+all the
+projects in a way that guarantees any project is built before it is required.
```
any --> each

elharo commented on content/markdown/guides/mini/guide-multiple-subprojects-4.md:
```diff
@@ -0,0 +1,184 @@
+
+Maven knows about the dependencies between subprojects inside the multi-project setup.
+Using `--also-make`, Maven includes all dependencies of the selected projects in the build.
+Similarly, `--also-make-dependents` lets Maven include all subprojects which are dependent on the selected projects.
```
are dependent --> depend

elharo commented on content/markdown/guides/mini/guide-multiple-subprojects-4.md:
```diff
@@ -0,0 +1,184 @@
+
+There are several ways to customize how the reactor deals with failures.
+`--fail-at-end` fails the build after building as many subprojects as possible.
+In this case, subprojects that do not depend on a failed subproject, will still be built.
```
subproject, will --> subproject will

elharo commented on content/markdown/guides/mini/guide-multiple-subprojects-4.md:
```diff
@@ -0,0 +1,183 @@
+
+### Collecting subprojects
+
+Subprojects collection starts from one aggregator project.
```
Subproject collection

Bukama commented:
Added @elharo suggestions and rebased (to resolve htaccess merge conflict)

jira-importer commented:
Resolve #1130
